packages/ui-vue/src/index-wc.ts

```diff
@@ -0,0 +1,57 @@
+import { createApp, defineCustomElement, getCurrentInstance, h } from 'vue';
+import OdkWebForm from './components/OdkWebForm.vue';
+import { webFormsPlugin } from './WebFormsPlugin';
+
+interface WebComponent {
+	styles: string[];
+	emits: string[];
+}
+
+const OdkWebFormWC = OdkWebForm as unknown as WebComponent;
```

Comment:
Really briefly (reiterating here, since we also chatted a bit about this earlier when we talked about the last change in #114)... this is a great example of what I've been talking about when I say that the various tools (ESLint, Vue, TypeScript, `typescript-eslint`... and all of their respective parsers that are meant to work together) fail to agree on whether a `.vue` component module does or does not have types.

The feedback that I **had written** was along these lines:

This will hopefully become less of an issue, when some [tooling improvements](https://github.com/volarjs/volar.js/pull/171) are fully available/when we're able to update the affected tooling.

Until that's resolved, I believe we should take a two pronged approach to the issue:

1. When we're merely importing a `.vue` component into a `.ts` module, and then passing it as a value to something else, I think we can generally accept the hit to type safety of a cast. This is what we've done so far.

2. When we're actually _interacting with the value_ of an imported `.vue` component (e.g. by accessing its properties as we are here, and passing them to APIs which expect them to be of a certain type)... I think we should take some more care to validate their types rather than casting them. Common approaches to this use [type predicates](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) and/or [assertion functions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions).

- - -

I also took some time, before posting those thoughts, to prototype some ideas around how we could do (2) in a general/reusable way that shouldn't be too onerous. I had an approach that felt like it was going in the right direction (stashed now, we can revisit). So I applied the concept to an import of `OdkWebForm.vue`, performing a runtime validation that the component does actually have an `emits` property which is an array of strings, and a `styles` property which is an array of strings.

But... the component does not have a `styles` property at all. All of the tooling issues aside, this type cast is incorrect.

My next hunch was that maybe that property becomes populated if a component specifically includes a `<style>` block. It doesn't in this PR, but I thought maybe it came up with #113 which does add a style block there. Unfortunately, at least adding a quick `<style>` block locally, I still do not see `styles: string[]` on the imported component. I also do not see anything else on the component which might provide the same data.

- - -

I don't really know what to suggest here. I don't know what the type should actually be, or what `styles` is supposed to contain.

Reply:
I have pushed a new commit which simplifies this a lot by taking your suggestion from the slack message:

> it looks like the body of the custom element definition's setup is analogous to a Vue component. i think...? would it maybe make sense if it were also in a .vue module

Reply:
Regarding missing `styles` property; it is the magic done by `@vitejs/plugin-vue`. When a `.vue` component is consider as custom element (by default when file extension is .ce.vue) then the plugin extracts the style text and adds it to the component. See this https://vuejs.org/guide/extras/web-components.html#sfc-as-custom-element